FirebaseFunctionsSwift/Tests/Unit/ContextProviderTests.swift

```diff
@@ -12,181 +12,140 @@
-//    [expectation fulfill];
-//  }];
-//
-//  [self waitForExpectations:@[ expectation ] timeout:0.1];
-// }
-//
-// - (void)testAllContextsAuthAndAppCheckError {
-//  self.appCheckFake.tokenResult = self.appCheckTokenError;
-//
-//  NSError *authError = [[NSError alloc] initWithDomain:@"com.functions.tests" code:4 userInfo:nil];
-//  FIRAuthInteropFake *auth = [[FIRAuthInteropFake alloc] initWithToken:nil
-//                                                                userID:nil
-//                                                                 error:authError];
-//
-//  FUNContextProvider *provider = [[FUNContextProvider alloc] initWithAuth:auth
-//                                                                messaging:self.messagingFake
-//                                                                 appCheck:self.appCheckFake];
-//  XCTestExpectation *expectation =
-//      [self expectationWithDescription:@"Completion handler should succeed without Auth."];
-//  [provider getContext:^(FUNContext *context, NSError *_Nullable error) {
-//    XCTAssertNotNil(context);
-//    XCTAssertEqual(error, auth.error);
-//
-//    XCTAssertNil(context.authToken);
-//    XCTAssert([context.FCMToken isEqualToString:self.messagingFake.FCMToken]);
-//    // Don't expect any token in the case of App Check error.
-//    XCTAssertNil(context.appCheckToken);
-//    [expectation fulfill];
-//  }];
-//
-//  [self waitForExpectations:@[ expectation ] timeout:0.1];
-// }
-//
-// @end
+import Foundation
+
+import FirebaseCore
+@testable import FirebaseFunctionsSwift
+
+import FirebaseAppCheckInterop
+import FirebaseAuthInterop
+import FirebaseMessagingInterop
+import SharedTestUtilities
+
+import XCTest
+
+class ContextProviderTests: XCTestCase {
+  let appCheckFake = FIRAppCheckFake()
+  let appCheckTokenError = FIRAppCheckTokenResultFake(token: "dummy token",
+                                                      error: NSError(
+                                                        domain: "testAppCheckError",
+                                                        code: -1,
+                                                        userInfo: nil
+                                                      ))
+  let appCheckTokenSuccess = FIRAppCheckTokenResultFake(token: "valid_token", error: nil)
+  let messagingFake = FIRMessagingInteropFake()
+
+  func testContextWithAuth() {
+    let auth = FIRAuthInteropFake(token: "token", userID: "userID", error: nil)
+    let provider = FunctionsContextProvider(auth: auth, messaging: messagingFake, appCheck: nil)
+    let expectation = expectation(description: "Context should have auth keys.")
+    provider.getContext { context, error in
+      XCTAssertNotNil(context)
+      XCTAssertEqual(context.authToken, "token")
+      XCTAssertEqual(context.fcmToken, self.messagingFake.fcmToken)
+      XCTAssertNil(error)
+      expectation.fulfill()
+    }
+    waitForExpectations(timeout: 1.5)
```

Comment:
Does this timeout need to be so high compared to the others?

Reply:
Direct port, but no reason to be different than the others.